src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -119,6 +119,28 @@ internal static bool HasUnsafeParameter(this Symbol member)
             return false;
         }
 
+        public static bool IsEventOrPropertyWithNonPublicAccessor(this Symbol symbol)
+        {
+            switch (symbol.Kind)
+            {
+                case SymbolKind.Property:
+                    var propertySymbol = (PropertySymbol)symbol;
+
+                    bool AccessorIsNotPublic(MethodSymbol accessor)
```

Comment:
Consider moving out of the switch statement, since used by both cases.
Also, consider using lower-case (`accessorIsNotPublic`) for local function.